### DISCUSSION TITLE
Control already has visual parent when using Image in InlineCollection and switch to other view and back

### DISCUSSION THREAD

**Discussion by JulanDeAlb** [AvaloniaUI/Avalonia#11542] [Q&A] [answered]
Hey, im trying to implement a Twitch like Chat. For this i need to have a Dynamic Collection of TextBlocks and Images. For binding i use an ObservableCollection<InlineCollection.
The methode to display the Message looks the following:
```
private async void DisplayMessage(Message message)
    {
        var inlineCollection = new InlineCollection();
        var parts = message.TextMessage.Split(" ", StringSplitOptions.RemoveEmptyEntries);

        var run = new Run(message.Time);
        run.Foreground = Brushes.Gray;
        inlineCollection.Add(run);

        run = new Run(message.Username);
        run.Foreground = message.Color;
        inlineCollection.Add(run);

        foreach (var part in parts)
        {
            if (Images.TryGetValue(part, out var imageUrl))
            {
                using var client = new HttpClient();
                var response = await client.GetAsync(imageUrl).ConfigureAwait(true);
                using var stream = await response.Content.ReadAsStreamAsync().ConfigureAwait(true);
                var image = new Image();
                image.Source = new Bitmap(stream);

                inlineCollection.Add(image);
            }
            else
            {
                var text = part;
                text += part.Equals(parts.Last()) ? "" : " ";
                inlineCollection.Add(new Run(text));
            }
        }

        if (MyCollection.Count >= 500)
        {
            MyCollection.RemoveAt(0);
        }

        MyCollection.Add(inlineCollection);
    }
```
    
Doing it this way works fine, until i switch to another view and back to the view containing the messages. The Application crashes with the Error: The control already has a visual parent.
On further investigation i noticed it only happens when adding new Images to the InlineCollection. Without adding the images it works.

The axaml i use is
```
<ItemsControl ItemsSource="{Binding MyCollection}">
                <ItemsControl.ItemsPanel>
                    <ItemsPanelTemplate>
                        <StackPanel Orientation="Vertical" VerticalAlignment="Bottom"/>
                    </ItemsPanelTemplate>
                </ItemsControl.ItemsPanel>
                <ItemsControl.ItemTemplate>
                    <DataTemplate>
                        <TextBlock TextWrapping="Wrap" Inlines="{Binding}" />
                    </DataTemplate>
                </ItemsControl.ItemTemplate>
            </ItemsControl>
```
            
I would really appreciate it if someone could help me solve this issue.
Incase its needed, the full code for this view and viewmodel is added.
[HomeViewModel.cs.txt](https://github.com/AvaloniaUI/Avalonia/files/11582813/HomeViewModel.cs.txt)
[HomeView.axaml.cs.txt](https://github.com/AvaloniaUI/Avalonia/files/11582814/HomeView.axaml.cs.txt)
[HomeView.axaml.txt](https://github.com/AvaloniaUI/Avalonia/files/11582815/HomeView.axaml.txt)

**Comment by JGKle** [comment]
Did you ever get this solved? I'm having a similar issue, posted here: https://github.com/AvaloniaUI/Avalonia/issues/14003

**Comment by JulanDeAlb** [comment, marked as answer]
After looking though issues with the Visual Parent and seeing #11543 it got fixed without seeing it in this discussion.
